app/cdap/components/hydrator/components/TopPanel/NameAndDescription/index.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  CancelButton,
+  CustomTooltip,
+  DescriptionTextField,
+  HydratorMetadata,
+  MetadataLeft,
+  NameTextField,
+  PipelineDescription,
+  PipelineName,
+  PipelineType,
+  SaveButton,
+  StyledIconSVG,
+} from './styles';
+
+interface IGlobalObj {
+  etlRealtime?: string;
+  etlDataStreams?: string;
+  etlBatchPipelines?: string[];
+}
+
+interface IMetadata {
+  description: string;
+  name: string;
+}
+
+interface IArtifact {
+  name: string;
+  scope: string;
+  version: string;
+}
+
+interface INameState {
+  metadata?: IMetadata;
+  artifact?: IArtifact;
+  viewSettings?: boolean;
+}
+
+interface INameAndDescriptionProps {
+  metadataExpanded?: boolean;
+  globals?: IGlobalObj;
+  state?: INameState;
+  invalidName?: string;
+  parsedDescription?: string;
+  saveMetadata?: (event: any) => void;
+  resetMetadata?: (event: any) => void;
+  openMetadata?: () => void;
+}
+
+export const NameAndDescription = ({
+  metadataExpanded,
+  globals,
+  state,
+  invalidName,
+  parsedDescription,
+  saveMetadata,
+  resetMetadata,
+  openMetadata,
+}: INameAndDescriptionProps) => {
+  const [name, setName] = useState(state.metadata.name);
+  const [description, setDescription] = useState(state.metadata.description);
+  const handleNameChange = (e) => {
+    setName(e.target.value);
+    state.metadata.name = e.target.value;
+  };
+  const handleDescriptionChange = (e) => {
+    setDescription(e.target.value);
+    state.metadata.description = e.target.value;
+  };
+  const onEnterOnMetadata = (event) => {
+    // Save when user hits ENTER key.
+    if (event.nativeEvent.keyCode === 13) {
+      saveMetadata(event);
+    } else if (event.nativeEvent.keyCode === 27) {
+      // Reset if the user hits ESC key.
+      resetMetadata(event);
+    }
+  };
+
+  const eTLBatch = globals.etlBatchPipelines.includes(state.artifact.name);
+  const eTLRealtime = state.artifact.name === globals.etlRealtime;
+  const sparkstreaming = state.artifact.name === globals.etlDataStreams;
+
+  return (
+    <HydratorMetadata
+      expanded={metadataExpanded}
+      onClick={!metadataExpanded && openMetadata}
```

Comment:
so you can't close this thing?

Reply:
Decided to move the onClickHandler to below